jax/_src/scipy/special.py

```diff
@@ -2550,6 +2557,11 @@ def hyp1f1(a: ArrayLike, b: ArrayLike, x: ArrayLike) -> Array:
   # evaluate lower values of hyp1f1 when a or b or both are > 60-80
   a, b, x = promote_args_inexact('hyp1f1', a, b, x)
 
+  dtype = lax.dtype(x)
+  if dtype not in (jnp.float32, jnp.float64):
```

Comment:
This is a breaking change, because the function currently accepts float16 inputs. We'd need a deprecation period if we did this; easiest thing is to remove this block.

Reply:
Okay, this is removed!